presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -544,7 +546,10 @@ private PlanRoot createLogicalPlanAndOptimize()
             queryPlan.set(plan);
             stateMachine.setPlanStatsAndCosts(plan.getStatsAndCosts());
             stateMachine.setPlanIdNodeMap(plan.getPlanIdNodeMap());
-            stateMachine.setPlanCanonicalInfo(getCanonicalInfo(getSession(), plan.getRoot(), planCanonicalInfoProvider));
+            List<CanonicalPlanWithInfo> canonicalPlanWithInfos = getSession().getRuntimeStats().profileNanos(
+                    GET_CANONICAL_INFO_TIME_NANOS,
+                    () -> getCanonicalInfo(getSession(), plan.getRoot(), planCanonicalInfoProvider));
```

Comment:
The getCanonicalInfo can be expensive and takes long time during my debug, add profiling for it.